### PR TITLE
DI-212 Mock DoS API Gateway

### PR DESCRIPTION
**Pull request by JackPlowman**
## Link to JIRA Ticket

- https://nhsd-jira.digital.nhs.uk/browse/DI-212

## Description

This PR is to add a Mock for the DoS API Gateway. With 2 new lambdas (using Docker images).

### Noteworthy Changes

- API Gateway to act as DoS API Gateway
- Authoriser Lambda (used by API Gateway)
- DoS API Gateway Lambda (processing for Change Request)
- Updated all terraform to use default tags
- Username and Password for API Gateway is stored in Secrets Manager

## Type of change

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

## Testing

Please tick the testing that has been completed

- [x] Unit tests
- [x] Component tests

## Developer Checklist

- [x] I have performed a self-review of my own code
- [x] I have run the [code formatting checks](../README.md#code-quality)
- [x] I have run the [code quality checks](../README.md#code-quality)
- [x] New code meets [standards](https://nhsd-confluence.digital.nhs.uk/display/DI/DI+Ways+of+Working) agreed by the team
- [x] Unit test code coverage is at or above 80%
- [x] New and existing unit tests pass locally with my changes
- [x] Tests have added that prove my fix is effective or that my feature works (Component tests)
- [x] I have made corresponding changes to the documentation
- [ ] I have cleaned down my environment (if created)

## Code Reviewer Checklist

- [x] I have run the unit tests and they run correctly
- [ ] I can confirm the changes have been tested or approved by a tester
- [ ] I can confirm no remaining infrastructure is left over from this branch

### CODE REVIEW

**Comment by ed-sparkes** [comment]
Great work @JackPlowman

**Comment by JackPlowman** [comment]
Michael and I just test that a change event can be received, run through the process, arrive at the mock, log the change request and pass back a response to the event sender

**Review by ed-sparkes — CHANGES_REQUESTED** [review]

File: application/dos_api_gateway/dos_api_gateway.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    Returns:
+        dict: Response to change request
+    """
+    event = loads(event["body"])
+    print(event)
+    change_request_response = {"dosChanges": []}
+    counter = 1
+    for row in event["changes"]:
+        change_request_response["dosChanges"].append({"changeId": str(counter) * 9})
```
[the remark this answers is not in the document]
Comment (JackPlowman) [reply application/dos_api_gateway/dos_api_gateway.py:21]: This a line  adds a changeId for every change that is received. For example if there is two changes (website and phone) this should return this `{"dosChanges": [{"changeId": "111111111"}, {"changeId": "222222222"}]}`

**Review by ed-sparkes — APPROVED** [review]

**Review by rohith1122 — APPROVED** [review]
1. Run the unit tests and all passed
2. Found a  print statement in dos_api_gateway.py . otherwise looking good

**Comment by JackPlowman** [comment]
> 1. Run the unit tests and all passed
> 2. Found a  print statement in dos_api_gateway.py . otherwise looking good

At the moment that is the expected behaviour for the demo so that change requests are logged

**Pull request merged**